components/robotsports/getball-intercept/tick.cpp

```diff
@@ -0,0 +1,140 @@
+// this file was produced by MRA-codegen.py from template_tick.cpp
+// with the intent of letting user add the implementation here
+
+// generated component header:
+#include "RobotsportsGetballIntercept.hpp"
+
+// dependent libraries:
+#include "geometry.hpp"
+
+using namespace MRA;
+
+// custom includes, if any
+// ...
+
+/**
+ * Calculates the intersection point of line given by pose1  and pose2, and the line perpendicular to (pose1,pose2) going
+ * through point pose_p.
+ *
+ * @param px
+ *            X of Point though which the perpendicular line must go
+ * @param py
+ *            Y of Point though which the perpendicular line must go
+ * @param p1x
+ *            X of One point on the line (p1,p2)
+ * @param p1y
+ *            Y of One point on the line (p1,p2)
+ * @param p2x
+ *            X of Other point on the line (p1,p2)
+ * @param p2y
+ *            Y ofOther point on the line (p1,p2)
+ * @return Intersection point D. Note that c.distanceTo(D) is the distance of c to the line (p1,p2).
+ */
+static MRA::Geometry::Position intersectPerpendicular(const MRA::Geometry::Position& r_pose1, const MRA::Geometry::Position& r_pose2,
```

Comment:
I try to put generic/useful stuff like this in a library, in this case it would nicely fit in geometry. Thoughts?

Reply:
indeed, geometry will be good option. Also conversions like rad2deg and deg2rad can be added in future